### ISSUE TITLE
Inference Problems

### ISSUE THREAD

**Issue by YY2968218397** [NVIDIA-Medtech/NV-Segment-CTMR#8]
I find a problem in inference，Contrast-enhanced CT,
1.I inference the Hepatic vessels(label:25) only,it looks good;
2.I inference the liver(label:1) and the Hepatic vessels(label:25) at the same time,the liver looks good,but the Hepatic vessels change too much,nearly disappear,only a small portion has been segement.

I followed the readme instructions,and read the vista3d paper, it talk about Binary segmentation.
But,did I do something wrong?Do you have any suggestions?

**Comment by heyufan1995** [comment]
Hi: 
The hepatic vessel is part of liver and they overlap. When we save the segmentation, we perform argmax thus liver mask will override hepatic vessel. Follow the readme here, you can change the https://github.com/NVIDIA-Medtech/NV-Segment-CTMR/blob/6388d8f381186508e7229fe4383302c231dd27a3/NV-Segment-CTMR/configs/inference.json#L146  to 
```json
{ 
  "_target_": "Activationsd",
  "sigmoid": true,
  "keys": "pred"
},
```   
The output segmentation file will thus be multichannel.

There are also postprocessing methods:

## 3. VISTA3D results postprocessing with [ShapeKit](https://arxiv.org/pdf/2506.24003)
VISTA3D is trained with binary segmentation, and may produce false positives due to weak false positive supervision. ShapeKit solves this problem with sophisticated postprocessing. ShapeKit requires segmentation mask for each class. VISTA3D by default performs argmax and collaps overlapping classes. Change the `monai.apps.vista3d.transforms.VistaPostTransformd` in `inference.json` to save each class segmentation as a separate channel. Then follow [ShapeKit](https://github.com/BodyMaps/ShapeKit) codebase for processing.
```json
{ 
  "_target_": "Activationsd",
  "sigmoid": true,
  "keys": "pred"
},
```

**Comment by YY2968218397** [comment]
Yes,it work! i get the right result. thank you very much!

i find another Inference problem, about multi-gpu-inference.

i still use the 
{
       "_target_": "monai.apps.vista3d.transforms.VistaPostTransformd",
      "keys": "pred"
 },

1.i set the mri_body to [1] in inference.json,to inference the liver

2.i use the commond:python -m monai.bundle run --config_file="['configs/inference.json', 'configs/batch_inference.json']" --input_dir="example/" --output_dir="example/" --modality MRI_BODY, the save mask is ok;

3.i use the commond:CUDA_VISIBLE_DEVICE=0,2,3 torchrun --nproc_per_node=3 --nnodes=1 -m monai.bundle run --config_file="['configs/inference.json', 'configs/batch_inference.json', 'configs/mgpu_inference.json']" --input_dir="example/" --output_dir="example/",the save mask is not ok.

it maybe only my problem in my machine,do you have this problem?

**Comment by heyufan1995** [comment]
Hi Thanks for pointing out.
there is a bug with mgpu_inference.json. Please change it to this. I will update the repo soon.
```{
    "device": "$torch.device('cuda:' + os.environ['LOCAL_RANK'])",
    "network": {
        "_target_": "torch.nn.parallel.DistributedDataParallel",
        "module": "$@network_def.to(@device)",
        "device_ids": [
            "@device"
        ]
    },
    "sampler": {
        "_target_": "DistributedSampler",
        "dataset": "@dataset",
        "even_divisible": false,
        "shuffle": false
    },
    "dataloader#sampler": "@sampler",
    "initialize": [
        "$import torch.distributed as dist",
        "$dist.is_initialized() or dist.init_process_group(backend='nccl')",
        "$torch.cuda.set_device(@device)",
        "$monai.utils.set_determinism(seed=123)",
        "$@checkpointloader(@evaluator)"
    ],
    "run": [
        "$@evaluator.run()"
    ],
    "finalize": [
        "$dist.is_initialized() and dist.destroy_process_group()"
    ]
}
```

**Comment by heyufan1995** [comment]
@YY2968218397 Updated repo with batch inference.
